### PR TITLE
fix(error): avoid phpunit crash by triggering side-effect via count on testcase

### PR DESCRIPTION
**Pull request by simPod**
@mfn this is a followup of https://github.com/sebastianbergmann/phpunit/issues/5866#issuecomment-2172429263

What do you think about the fix? I don't see any other option since we analyze the whole stacktrace.

### CODE REVIEW

File: src/Error/FormattedError.php [hunk 1]
```diff
@@ -299,7 +300,11 @@ public static function printVar($var): string
         }
 
         if (\is_object($var)) {
-            return 'instance of ' . \get_class($var) . ($var instanceof \Countable ? '(' . \count($var) . ')' : '');
+            $count = ! $var instanceof Test && $var instanceof \Countable
```
Comment (spawnia) [inline src/Error/FormattedError.php:303]:
```suggestion
            // Calling `count` on PHPUnit\Framework\Test triggers an unintended side effect - see https://github.com/sebastianbergmann/phpunit/issues/5866#issuecomment-2172429263
            $count = ! $var instanceof Test && $var instanceof \Countable
```

File: src/Error/FormattedError.php [hunk 2]
```diff
@@ -299,7 +300,12 @@ public static function printVar($var): string
         }
 
         if (\is_object($var)) {
-            return 'instance of ' . \get_class($var) . ($var instanceof \Countable ? '(' . \count($var) . ')' : '');
+            // Calling `count` on instances of `PHPUnit\Framework\Test` triggers an unintended side effect - see https://github.com/sebastianbergmann/phpunit/issues/5866#issuecomment-2172429263
+            $count = ! $var instanceof Test && $var instanceof \Countable
```
Comment (spawnia) [inline src/Error/FormattedError.php:304]: How does this behave when PHPUnit is not installed?

Comment (simPod) [reply src/Error/FormattedError.php:304]: I've tested it and it seems to be just k https://3v4l.org/FFKFf

**Comment by spawnia** [comment]
Thank you @simPod and @mfn! Released with https://github.com/webonyx/graphql-php/releases/tag/v15.12.3.

**Pull request merged**